grails-app/domain/org/pih/warehouse/core/Synonym.groovy

```diff
@@ -11,6 +11,7 @@ package org.pih.warehouse.core
 
 import org.pih.warehouse.auth.AuthService
 import org.pih.warehouse.product.Product
+import org.pih.warehouse.product.SynonymClassification
```

Comment:
Synonym is in the core package (i.e. we could add a synonyms association to other objects as well). So this should also be in the core package. To be more like other enums it should probably be SynonymTypeCode. We may also want to create a SynonymType but that might be overkill at the moment.